### PR TITLE
require appropriate content-type header when sending content to server

### PR DESCRIPTION
**Pull request by bobholt**
- require 'application/vnd.api+json' contentType header in create/update
- send 415 error if not included

### CODE REVIEW

File: modules/controller/lib/verify_content_type.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+module.exports = function(request) {
+  var contentType = 'application/vnd.api+json';
+  return (
+    request.header('content-type') &&
```
Comment (tkellen) [inline modules/controller/lib/verify_content_type.js]: If you use `request.headers['content-type']` this will work for both hapi and express.

**Comment by tkellen** [comment]
I'm hesitant to pile even more closures on to the controller methods, but would it make sense to abstract the function that generates the request handlers to do this automatically?

**Comment by tkellen** [comment]
Something like
```js
var requestHandler = function (opts, callback) {
  return function (request, response) {
    // check headers here based on opts
    callback(request, response);
  }
};
```

**Comment by bobholt** [comment]
I think there's room for doing something like that. I feel like I'm so close to an epiphany about the way to refactor those controller methods. It might involve something like that?

**Comment by tkellen** [comment]
I think you're right!  I was feeling the same thing.

**Comment by bobholt** [comment]
This is going to get wrapped up in a refactor

**Pull request closed without merging**